package-lock.json

```diff
@@ -53051,1109 +51467,145 @@
-			"version": "7.25.7",
-			"resolved": "https://registry.npmjs.org/@babel/plugin-transform-classes/-/plugin-transform-classes-7.25.7.tgz",
-			"integrity": "sha512-9j9rnl+YCQY0IGoeipXvnk3niWicIB6kCsWRGLwX241qSXpbA4MKxtp/EdvFxsc4zI5vqfLxzOd0twIJ7I99zg==",
-			"dev": true,
-			"license": "MIT",
+		"packages/block-editor": {
+			"name": "@wordpress/block-editor",
+			"version": "14.4.0",
+			"license": "GPL-2.0-or-later",
 			"dependencies": {
-				"@babel/helper-annotate-as-pure": "^7.25.7",
-				"@babel/helper-compilation-targets": "^7.25.7",
-				"@babel/helper-plugin-utils": "^7.25.7",
-				"@babel/helper-replace-supers": "^7.25.7",
-				"@babel/traverse": "^7.25.7",
-				"globals": "^11.1.0"
+				"@babel/runtime": "7.25.7",
+				"@emotion/react": "^11.7.1",
+				"@emotion/styled": "^11.6.0",
+				"@react-spring/web": "^9.4.5",
+				"@wordpress/a11y": "*",
+				"@wordpress/api-fetch": "*",
+				"@wordpress/blob": "*",
+				"@wordpress/block-serialization-default-parser": "*",
+				"@wordpress/blocks": "*",
+				"@wordpress/commands": "*",
+				"@wordpress/components": "*",
+				"@wordpress/compose": "*",
+				"@wordpress/data": "*",
+				"@wordpress/date": "*",
+				"@wordpress/deprecated": "*",
+				"@wordpress/dom": "*",
+				"@wordpress/element": "*",
+				"@wordpress/escape-html": "*",
+				"@wordpress/hooks": "*",
+				"@wordpress/html-entities": "*",
+				"@wordpress/i18n": "*",
+				"@wordpress/icons": "*",
+				"@wordpress/is-shallow-equal": "*",
+				"@wordpress/keyboard-shortcuts": "*",
+				"@wordpress/keycodes": "*",
+				"@wordpress/notices": "*",
+				"@wordpress/preferences": "*",
+				"@wordpress/private-apis": "*",
+				"@wordpress/rich-text": "*",
+				"@wordpress/style-engine": "*",
+				"@wordpress/token-list": "*",
+				"@wordpress/url": "*",
+				"@wordpress/warning": "*",
+				"@wordpress/wordcount": "*",
+				"change-case": "^4.1.2",
+				"clsx": "^2.1.1",
+				"colord": "^2.7.0",
+				"deepmerge": "^4.3.0",
+				"diff": "^4.0.2",
+				"fast-deep-equal": "^3.1.3",
+				"memize": "^2.1.0",
+				"parsel-js": "^1.1.2",
+				"postcss": "^8.4.21",
+				"postcss-prefix-selector": "^1.16.0",
+				"postcss-urlrebase": "^1.4.0",
+				"react-autosize-textarea": "^7.1.0",
+				"react-easy-crop": "^5.0.6",
+				"remove-accents": "^0.5.0"
 			},
 			"engines": {
-				"node": ">=6.9.0"
+				"node": ">=18.12.0",
+				"npm": ">=8.19.2"
 			},
 			"peerDependencies": {
-				"@babel/core": "^7.0.0-0"
+				"react": "^18.0.0",
+				"react-dom": "^18.0.0"
 			}
 		},
-		"packages/babel-preset-default/node_modules/@babel/plugin-transform-computed-properties": {
-			"version": "7.25.7",
-			"resolved": "https://registry.npmjs.org/@babel/plugin-transform-computed-properties/-/plugin-transform-computed-properties-7.25.7.tgz",
-			"integrity": "sha512-QIv+imtM+EtNxg/XBKL3hiWjgdLjMOmZ+XzQwSgmBfKbfxUjBzGgVPklUuE55eq5/uVoh8gg3dqlrwR/jw3ZeA==",
-			"dev": true,
+		"packages/block-editor/node_modules/postcss-urlrebase": {
```

Comment:
Finally, I see a lot of packages hoisted after changes are applied with this PR, but there are still some places where npm packages are installed inside the local package. I hope we can remove all these instances over time.

Reply:
Some duplication is likely inevitable because it's very difficult to get all transitive dependencies onto the same version of a package all the time. When incompatible versions manifest, I think that's always going to cause deps to be installed under packages.

Hopefully this will get us to a more sane place of dependency management. This is in a place where I can try one of the dependency upgrades that was constantly stuck and see if it works better (e.g. #61628).